File: HW1 Cpp/outdoor.cpp
//In this code , I define  implementation of the OutdoorSports class methods.
// I added constructor to initialize the object with values for name, price, stock, size, durabilit and weather rating
// Constructor to initialize the OutdoorSports object by passing the parameters to our base class Product
#include "outdoor.h"
#include <iostream>
OutdoorSports::OutdoorSports(string n, double p, int s, int size, string durability, int weatherRating)
: Product(n, p, s), size(size), durability(durability), weatherRating(weatherRating) {}
void OutdoorSports::getDetails() {
    cout << name << " (Outdoor - Size: " << size << ", Durability: " << durability << ", Weather Rating: " << weatherRating << "): $" << price << ", Stock: " << stock << endl;
}

File: HW1 Py/outdoor.py
from product import Product
# In this part I created Outdoor derived class from Product base claas and added some additinal parameters
class OutdoorProduct(Product):
    def __init__(self, name, price, stock, size, durability, weather_rating):
        super().__init__(name, price, stock)
        self.size = size
        self.durability = durability
        self.weather_rating = weather_rating
    
    def __str__(self):
        return f"{self.name} (Outdoor - Size: {self.size}, Durability: {self.durability}, Weather Rating: {self.weather_rating}): ${self.price}, Stock: {self.stock}"